backend/src/org/akvo/lumen/component/transformation_engine.clj

```diff
@@ -11,25 +11,26 @@
   (start [this]
     (reset! running? true)
     (let [engine (future
-                   (loop [;; TODO naming?
-                          transformation (.take queue)]
-                     (when-not (identical? transformation sentinel)
-                       (let [{:keys [completion-promise tenant-conn job-id dataset-id command]} transformation]
-                         (condp = (:type command)
-                           :transformation
-                           (engine/execute-transformation completion-promise
-                                                          tenant-conn
-                                                          job-id
-                                                          dataset-id
-                                                          (:transformation command))
+                   (loop [transformation (.take queue)]
+                     (try
+                       (when-not (identical? transformation sentinel)
+                         (let [{:keys [completion-promise tenant-conn job-id dataset-id command]} transformation]
+                           (condp = (:type command)
+                             :transformation
+                             (engine/execute-transformation completion-promise
+                                                            tenant-conn
+                                                            job-id
+                                                            dataset-id
+                                                            (:transformation command))
 
```

Comment:
I don't mind either but the second arg on new line I find a bit difficult to parse without counting parens. Like:
```
(cond test-a
  test-b)
```